src/state/session/index.tsx

```diff
@@ -392,97 +408,34 @@ export function Provider({children}: React.PropsWithChildren<{}>) {
         logger.error(`session: could not decode jwt`)
       }
 
-      const prevSession = {
-        accessJwt: account.accessJwt || '',
-        refreshJwt: account.refreshJwt || '',
-        did: account.did,
-        handle: account.handle,
-        deactivated:
-          isSessionDeactivated(account.accessJwt) || account.deactivated,
-      }
+      // optimistic, we'll update this if we can't reuse or resume the session
+      await configureModeration(agent, account)
 
       if (canReusePrevSession) {
```

Comment:
This block is just simplified.

If we know the session hasn't expired:
- set it on `agent.session`
- update `__globalAgent`
- update state, trigger app rerender

If the session is expired:
- check if it can be refreshed
  - if it succeeds, the persist callback will handle the rest
- if it fails, clear the current account